### DISCUSSION TITLE
External logins - where are the additional authentication schemes registered?

### DISCUSSION THREAD

**Discussion by weirdyang** [OrchardCMS/OrchardCore#14977] [Q&A] [answered]
I notice in the modules for the Facebook and Twitter logins, the call to AddFacebook or AddTwitter isn't made in the startup.cs, how are these schemes registered? I don't see them in the users module either.

**Comment by MichaelPetrinolis** [comment]
Check the startup files. Eg. TwitterSigninStartup registers all the services needed.

**Reply by weirdyang** [reply, marked as answer]
Thanks it is under the configuration files, not in startup.cs

https://github.com/OrchardCMS/OrchardCore/blob/42837c9ba0ad0bc86728c358926d4cde72620eff/src/OrchardCore.Modules/OrchardCore.Twitter/Signin/Configuration/TwitterOptionsConfiguration.cs#L62
